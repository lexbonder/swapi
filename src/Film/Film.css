.Film h3 {
  font-weight: normal;
}

h3 {
  font-size: 26px;
  margin: 10px 0;
  text-align: center;
}