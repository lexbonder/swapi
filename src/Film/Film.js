import React from 'react';
import PropTypes from 'prop-types';

const Film = ({title, date}) => {
  return (
    <li>{title} - {date}</li>
  );
};

const { string } = PropTypes;

Film.propTypes = {
  title: string,
  date: string
};

export default Film;