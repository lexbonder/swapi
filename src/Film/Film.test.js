/* eslint-disable */
import React from 'react';
import Film from './Film';
import { shallow } from 'enzyme';

describe('Film', () => {
  let wrapper;
  let mockTitle = 'Empire Strikes Back';
  let mockDate = "Saturday, May 17th 1980";

  beforeEach(() => {
    wrapper = shallow(<Film 
      title={mockTitle}
      date={mockDate}
    />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})