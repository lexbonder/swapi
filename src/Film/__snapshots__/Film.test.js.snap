// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Film
    date="Saturday, May 17th 1980"
    title="Empire Strikes Back"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Empire Strikes Back",
        " - ",
        "Saturday, May 17th 1980",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Empire Strikes Back",
      " - ",
      "Saturday, May 17th 1980",
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Empire Strikes Back",
          " - ",
          "Saturday, May 17th 1980",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Empire Strikes Back",
        " - ",
        "Saturday, May 17th 1980",
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
